tests/test_integration.py

```diff
@@ -14,39 +21,54 @@
 
 import docker
 
-REPO_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
-AF_ROOT = os.path.join(REPO_ROOT, 'af')
-TESTDATA_DIR = os.path.join(REPO_ROOT, 'testdata')
-
-METADB_NAME = 'ootestmetadb'
-METADB_PG_USER = 'oopguser'
-
-def start_pg(client):
-    pg_container = client.containers.run(
-            'postgres:9.6',
-            name=METADB_NAME,
-            hostname=METADB_NAME,
-            environment={
-                'POSTGRES_USER': METADB_PG_USER
-            },
-            ports={
-                '5432/tcp': 25432
-            },
-            detach=True
+REPO_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+AF_ROOT = os.path.join(REPO_ROOT, "af")
+TESTDATA_DIR = os.path.join(REPO_ROOT, "testdata")
+
+METADB_NAME = "ootestmetadb"
+METADB_PG_USER = "oopguser"
+
+
+conf = None
+
+
+def start_postgres_container(docker_client):
+    ## Start Postgres in a dedicated container named in METADB_NAME
+    running = docker_client.containers.list(filters=dict(name=METADB_NAME))
+    assert len(running) < 2
+    if len(running) == 1:
+        print("Postgres container already running: removing it")
+        running[0].stop()
+        running[0].remove()
+
+    print("Starting Postgres container")
+    pg_container = docker_client.containers.run(
+        "postgres:9.6",
+        name=METADB_NAME,
+        hostname=METADB_NAME,
+        environment={"POSTGRES_USER": METADB_PG_USER},
+        ports={"5432/tcp": 25432},
+        detach=True,
     )
     while True:
```

Comment:
Meta thought: do we care about this potentially running forever?

Reply:
It hasn't been a problem so far but a better try/finally to stop it even on test failure would be good.

Reply:
Maybe we can add a TODO comment saying that we can make this change in the future?

Reply:
Added a comment.